### PR TITLE
[Merged by Bors] - Implement resource requests and limits for Druid pods

### PR DESCRIPTION
**Pull request by maltesander**
# Description

closes https://github.com/stackabletech/druid-operator/issues/277

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- PVCs for data storage, cpu and memory limits are now configurable ([#298]).
```
Comment (sbernauer) [inline CHANGELOG.md:9]:
```suggestion
- Cpu and memory limits are now configurable ([#298]).
```

**Comment by maltesander** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/12/

File: tests/templates/kuttl/resources/00-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+metadata:
+  name: install-zk
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/00-assert.yaml:5]:
```suggestion
```

File: tests/templates/kuttl/resources/00-install-zk.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+apiVersion: zookeeper.stackable.tech/v1alpha1
+kind: ZookeeperCluster
+metadata:
+  name: druid-zk
+spec:
+  servers:
+    roleGroups:
+      default:
+        replicas: 1
+  version: {{ test_scenario['values']['zookeeper'] }}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/00-install-zk.yaml.j2:11]: I don't see a benefit testing against all zk versions. I suggest to use (or introduce) zookeeper-latest

Comment (maltesander) [reply tests/templates/kuttl/resources/00-install-zk.yaml.j2:11]: yeah thought about that, will do.

Comment (sbernauer) [reply tests/templates/kuttl/resources/00-install-zk.yaml.j2:11]: :+1:

Comment (sbernauer) [inline tests/templates/kuttl/resources/00-install-zk.yaml.j2:11]:
```suggestion
  version: {{ test_scenario['values']['zookeeper-latest'] }}
```

**Review by sbernauer — APPROVED** [review]
LGTM % comment when tests succeed

**Comment by maltesander** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/13/ another try (resources worked, two others failed)

**Review by sbernauer — APPROVED** [review]
Nice thanks!

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**